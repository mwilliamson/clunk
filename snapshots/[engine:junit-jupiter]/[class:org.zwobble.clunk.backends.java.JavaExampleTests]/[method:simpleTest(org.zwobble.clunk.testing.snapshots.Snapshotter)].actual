Source path: src/SimpleTest.clunk
import stdlib/assertions.assertThat;
import stdlib/matchers.equalTo;

test "assert False" {
    assertThat(false, equalTo(true));
}

test "assert True" {
    assertThat(true, equalTo(false));
}

test "assert 42 == 47" {
    assertThat(42, equalTo(47));
}

test "assert 42 == 42" {
    assertThat(42, equalTo(42));
}


==============

Output path: SimpleTest/SimpleTest.java
package SimpleTest;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;

public class SimpleTest {
    @org.junit.jupiter.api.Test
    @org.junit.jupiter.api.DisplayName("assert False")
    public void assertFalse() {
        assertThat(false, equalTo(true));
    }
    @org.junit.jupiter.api.Test
    @org.junit.jupiter.api.DisplayName("assert True")
    public void assertTrue() {
        assertThat(true, equalTo(false));
    }
    @org.junit.jupiter.api.Test
    @org.junit.jupiter.api.DisplayName("assert 42 == 47")
    public void assert42Equals47() {
        assertThat(42, equalTo(47));
    }
    @org.junit.jupiter.api.Test
    @org.junit.jupiter.api.DisplayName("assert 42 == 42")
    public void assert42Equals42() {
        assertThat(42, equalTo(42));
    }
}

==============

╷
├─ JUnit Jupiter ✔
│  └─ SimpleTest ✔
│     ├─ assert 42 == 42 ✔
│     ├─ assert 42 == 47 ✘ 
│     │        Expected: <47>
│     │             but: was <42>
│     ├─ assert False ✘ 
│     │        Expected: <true>
│     │             but: was <false>
│     └─ assert True ✘ 
│              Expected: <false>
│                   but: was <true>
└─ JUnit Vintage ✔

Failures (3):
  JUnit Jupiter:SimpleTest:assert 42 == 47
    MethodSource [className = 'SimpleTest.SimpleTest', methodName = 'assert42Equals47', methodParameterTypes = '']
    => java.lang.AssertionError: 
Expected: <47>
     but: was <42>
       org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
       org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
       SimpleTest.SimpleTest.assert42Equals47(Unknown Source)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
       java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
       java.base/java.lang.reflect.Method.invoke(Method.java:568)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]
  JUnit Jupiter:SimpleTest:assert False
    MethodSource [className = 'SimpleTest.SimpleTest', methodName = 'assertFalse', methodParameterTypes = '']
    => java.lang.AssertionError: 
Expected: <true>
     but: was <false>
       org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
       org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
       SimpleTest.SimpleTest.assertFalse(Unknown Source)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
       java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
       java.base/java.lang.reflect.Method.invoke(Method.java:568)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]
  JUnit Jupiter:SimpleTest:assert True
    MethodSource [className = 'SimpleTest.SimpleTest', methodName = 'assertTrue', methodParameterTypes = '']
    => java.lang.AssertionError: 
Expected: <false>
     but: was <true>
       org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
       org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
       SimpleTest.SimpleTest.assertTrue(Unknown Source)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
       java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
       java.base/java.lang.reflect.Method.invoke(Method.java:568)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       [...]

Test run finished after 72 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         4 tests found           ]
[         0 tests skipped         ]
[         4 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         3 tests failed          ]
