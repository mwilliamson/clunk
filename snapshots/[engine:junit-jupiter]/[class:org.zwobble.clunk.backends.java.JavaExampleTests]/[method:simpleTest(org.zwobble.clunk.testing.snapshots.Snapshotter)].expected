Source path: src/simple.test.clunk
import stdlib/assertions.assertThat;
import stdlib/matchers.equalTo;

test "assert False" {
    assertThat(false, equalTo(true));
}

test "assert True" {
    assertThat(true, equalTo(false));
}

test "assert 42 == 47" {
    assertThat(42, equalTo(47));
}

test "assert 42 == 42" {
    assertThat(42, equalTo(42));
}


==============

Output path: simple/SimpleTests.java
package simple;

import static org.zwobble.precisely.Matchers.equalTo;
import static org.zwobble.precisely.AssertThat.assertThat;

public class SimpleTests {
    @org.junit.jupiter.api.Test
    @org.junit.jupiter.api.DisplayName("assert False")
    public void assertFalse() {
        assertThat(false, equalTo(true));
    }
    @org.junit.jupiter.api.Test
    @org.junit.jupiter.api.DisplayName("assert True")
    public void assertTrue() {
        assertThat(true, equalTo(false));
    }
    @org.junit.jupiter.api.Test
    @org.junit.jupiter.api.DisplayName("assert 42 == 47")
    public void assert42Equals47() {
        assertThat(42, equalTo(47));
    }
    @org.junit.jupiter.api.Test
    @org.junit.jupiter.api.DisplayName("assert 42 == 42")
    public void assert42Equals42() {
        assertThat(42, equalTo(42));
    }
}

==============

╷
├─ JUnit Jupiter ✔
│  └─ SimpleTests ✔
│     ├─ assert 42 == 42 ✔
│     ├─ assert 42 == 47 ✘ 
│     │        Expected:
│     │          47
│     │        but:
│     │          was 42
│     ├─ assert False ✘ 
│     │        Expected:
│     │          true
│     │        but:
│     │          was false
│     └─ assert True ✘ 
│              Expected:
│                false
│              but:
│                was true
└─ JUnit Vintage ✔

Failures (3):
  JUnit Jupiter:SimpleTests:assert 42 == 47
    MethodSource [className = 'simple.SimpleTests', methodName = 'assert42Equals47', methodParameterTypes = '']
    => java.lang.AssertionError: 
Expected:
  47
but:
  was 42
       org.zwobble.precisely.AssertThat.assertThat(AssertThat.java:12)
       simple.SimpleTests.assert42Equals47(Unknown Source)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
       java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
       java.base/java.lang.reflect.Method.invoke(Method.java:568)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
       [...]
  JUnit Jupiter:SimpleTests:assert False
    MethodSource [className = 'simple.SimpleTests', methodName = 'assertFalse', methodParameterTypes = '']
    => java.lang.AssertionError: 
Expected:
  true
but:
  was false
       org.zwobble.precisely.AssertThat.assertThat(AssertThat.java:12)
       simple.SimpleTests.assertFalse(Unknown Source)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
       java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
       java.base/java.lang.reflect.Method.invoke(Method.java:568)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
       [...]
  JUnit Jupiter:SimpleTests:assert True
    MethodSource [className = 'simple.SimpleTests', methodName = 'assertTrue', methodParameterTypes = '']
    => java.lang.AssertionError: 
Expected:
  false
but:
  was true
       org.zwobble.precisely.AssertThat.assertThat(AssertThat.java:12)
       simple.SimpleTests.assertTrue(Unknown Source)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
       java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
       java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
       java.base/java.lang.reflect.Method.invoke(Method.java:568)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
       [...]

Test run finished after TIME ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         4 tests found           ]
[         0 tests skipped         ]
[         4 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         3 tests failed          ]
